### PR TITLE
Homework4

### PR DESCRIPTION
**Pull request by sls88**

### CODE REVIEW

File: hw/hw4/Homework_4_task1.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    Args:
+        path: file_path.txt
+
+    Returns:
+        The return value. True if number in an interval [1, 3), None if file is empty
+    """
+    try:
+        with open(path) as file:
+            for stri in file:
```
Comment (Aleksei-Samarin) [inline hw/hw4/Homework_4_task1.py]: Here we can delete nested block (reading data with further usage)

File: hw/hw4/Homework_4_task2.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    Args:
+        url: URL
+
+    Returns:
+        The return value. Amount of letters `i`
+    """
+    try:
+        res = requests.get(url)
+    except requests.exceptions.RequestException:
```
Comment (Aleksei-Samarin) [inline hw/hw4/Homework_4_task2.py]: Can be simplified by importing RequestException

File: hw/hw4/Homework_4_task3.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    """Receive a string and write it to stderr if line starts with "error" and to the stdout otherwise.
+
+    Args:
+        text: text
+
+    Returns:
+        The return value. text in stderr if line starts with "error", in stdout otherwise
+    """
+    if text.startswith("error"):
+        sys.stderr.write(text)
```
Comment (Aleksei-Samarin) [inline hw/hw4/Homework_4_task3.py:15]: These two conditions differs only on one object. Can we solve this with 2 lines?

File: requirements-dev.txt [hunk 4]
```diff
@@ -19,3 +19,9 @@ pre-commit==2.9.3
 
 pytest>=6.2.1
 pytest-cov>=2.10.1
+
+requests
```
Comment (Aleksei-Samarin) [inline requirements-dev.txt]: Add here only exact compatible versions without intervals

File: hw/hw4/Homework_4_task5.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+        Returns:
+            The return value. Element of FizzBuzz sequence
+        """
+        lst = [
+            list(
+                map(
+                    str,
+                    [
+                        1 + i * 15,
```
Comment (Aleksei-Samarin) [inline hw/hw4/Homework_4_task5.py]: Can we generate it programatically?

**Review by Aleksei-Samarin — CHANGES_REQUESTED** [review]

File: hw/hw4/Homework_4_task2.py (hunk 2 above)
Comment (AlbertGaz) [reply hw/hw4/Homework_4_task2.py:16]: Did you mean somethong like this?

dataflow = sys.stderr if text.startswith("error") else sys.stdout
dataflow.write(text)

File: requirements-dev.txt [hunk 6]
```diff
@@ -20,8 +20,8 @@ pre-commit==2.9.3
 pytest>=6.2.1
 pytest-cov>=2.10.1
 
-requests
-certifi>=2020.6.20
-chardet<4,>=3.0.4
-idna<3,>=2.10
-urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.11
+requests==2.24.0
+certifi==2020.6.20
```
Comment (Aleksei-Samarin) [inline requirements-dev.txt]: Do we need other packages except requests or it includes all of the rest packages?

**Review by Aleksei-Samarin — CHANGES_REQUESTED** [review]

**Pull request merged**